website/static/js/registrationUtils.js

```diff
@@ -0,0 +1,764 @@
+        if (question.required) {
+            var valid = question.value.isValid();
+            proceed = proceed && valid;
+        }
+    });
+    if (!proceed) {
+
+        bootbox.dialog({
+            title: 'Registration Not Complete',
+            message: 'There are errors in your registration. Please double check it and submit again.',
+            buttons: {
+                success: {
+                    label: 'Ok',
+                    className: 'btn-success',
+                    callback: function() {
+                        self.showValidation(true);
+                    }
+                }
+            }
+        });
+    } else {
+        window.location = self.draft().urls.register_page;
+    }
+};
+/**
+ * Load the next question into the editor, wrapping around if needed
+ **/
+RegistrationEditor.prototype.nextQuestion = function() {
+    var self = this;
+
+    var currentQuestion = self.currentQuestion();
+
+    var questions = self.flatQuestions();
+    var index = $osf.indexOf(questions, function(q) {
+        return q.id === currentQuestion.id;
+    });
+    if (index + 1 === questions.length) {
+        self.currentQuestion(questions.shift());
+    } else {
+        self.currentQuestion(questions[index + 1]);
+    }
+};
+/**
+ * Load the previous question into the editor, wrapping around if needed
+ **/
+RegistrationEditor.prototype.previousQuestion = function() {
+    var self = this;
+
+    var currentQuestion = self.currentQuestion();
+
+    var questions = self.flatQuestions();
+    var index = $osf.indexOf(questions, function(q) {
+        return q.id === currentQuestion.id;
+    });
+    if (index - 1 < 0) {
+        self.currentQuestion(questions.pop());
+    } else {
+        self.currentQuestion(questions[index - 1]);
+    }
+};
+/**
+ * Select a page, selecting the first question on that page
+ **/
+RegistrationEditor.prototype.selectPage = function(page) {
+    var self = this;
+
+    var firstQuestion = page.questions[Object.keys(page.questions)[0]];
+    self.currentQuestion(firstQuestion);
+};
+/**
+ * Update draft primary key and updated time on server response
+ **/
+RegistrationEditor.prototype.updateData = function(response) {
+    var self = this;
+
+    var draft = self.draft();
+    draft.pk = response.pk;
+    draft.updated = new Date(response.updated);
+    self.draft(draft);
+};
+RegistrationEditor.prototype.submitForReview = function() {
+    var self = this;
+
+    var draft = self.draft();
+    var metaSchema = draft.metaSchema;
+    var messages = metaSchema.messages;
+    var beforeSubmitForApprovalMessage = messages.beforeSubmitForApproval || '';
+    var afterSubmitForApprovalMessage = messages.afterSubmitForApproval || '';
+
+    bootbox.confirm(beforeSubmitForApprovalMessage, function(confirmed) {
+        if (confirmed) {
+            $osf.postJSON(self.urls.submit.replace('{draft_pk}', self.draft().pk), {}).then(function() {
```

Comment:
Also, for readability, please only use one method per line when chaining methods. (or just set `var request = ...; request.done; request.fail` separately)

https://github.com/felixge/node-style-guide#method-chaining